Kernel_23/include/CGAL/Bbox_2.h

```diff
@@ -155,7 +157,17 @@ Bbox_2::operator+=(const Bbox_2& b)
   rep[3] = (std::max)(ymax(), b.ymax());
   return *this;
 }
-
+inline
+void
+Bbox_2::dilate(int dist)
+{
+  using boost::math::float_advance;
+  float_advance(rep[0],-dist);
+  float_advance(rep[1],-dist);
+  float_advance(rep[2],dist);
+  float_advance(rep[3],dist);
+}
```

Comment:
`float_advance` does not modify its arguments, but returns a value!

That is why the bug #2999 is still there.